packages/core/admin/webpack.config.js

```diff
@@ -47,6 +48,20 @@ module.exports = ({
       ]
     : [];
 
+  // webpack is quite slow to compile so it is best not to use the ts loader when
+  // it is not needed in javascript apps.
+  // Users can still add it by using the custom webpack config.
+  const typescriptRules = useTypeScript
+    ? [
+        {
+          test: /\.tsx?$/,
+          use: require.resolve('ts-loader'),
+          include: [cacheDir, ...pluginsPath],
+          exclude: /node_modules/,
+        },
+      ]
+    : [];
+
```

Comment:
To speedup typescript build [`esbuild-loaded`](https://github.com/privatenumber/esbuild-loader#typescript--tsx) can be used instead of `ts-loader`